cpp/tests/utilities/base_fixture.hpp

```diff
@@ -167,5 +164,105 @@ class TempDirTestEnvironment : public ::testing::Environment {
   }
 };
 
+
+/**---------------------------------------------------------------------------*
+ * @brief Test environment that initializes the default rmm memory resource.
+ * 
+ * Required for tests programs that use rmm. It is recommended to include
+ * `CUDF_TEST_PROGRAM_MAIN()` in a code file instead of directly instantiating 
+ * an object of this type.
+ *---------------------------------------------------------------------------**/
+class RmmTestEnvironment : public ::testing::Environment {
+/**---------------------------------------------------------------------------*
+ * @brief String representing which RMM allocation mode is to be used.
+ *
+ * Valid values are 'cuda', 'pool' and 'managed'. 
+ *---------------------------------------------------------------------------**/
+  std::string mode;
+public:
+  /**
+   * @brief Constructor where the only argument is a string `mode`.
+   *
+   * Mode repesents the rmm allocation mode, @ref mode.
+   */
+  RmmTestEnvironment(std::string const& mode):mode(mode){}
+
+  /**
+   * @brief Initializes the default RMM memory resource.
+   *
+   * Sets different allocation mode based on the value passed to the constructor.
+   * 
+   * @throws cudf::logic_error if mode value is invalid.
+   * 
+   * @return void
+   */
+  void SetUp() {
+    rmmOptions_t opts{};
+    if (mode == "cuda")
+      opts.allocation_mode = CudaDefaultAllocation;
+    else if (mode == "pool")
+      opts.allocation_mode = PoolAllocation;
+    else if (mode == "managed")
+      opts.allocation_mode = CudaManagedMemory;
+    else 
+      CUDF_FAIL("Invalid RMM allocation mode");
+
+    ASSERT_EQ(rmmInitialize(&opts), RMM_SUCCESS);
+  }
```

Comment:
If 
> Sets different allocation mode based on the value passed to the constructor.

how come we can't just set this in the ctor?

Reply:
We can, but my understanding is that using `SetUp` is the idiomatic approach. 
If you have multiple test environments, the gtest framework ensures that SetUp is called in the order in which they are registered, and the TearDown is done in reverse order. This help with environment dependencies that we might have in the future.

Reply:
My understanding that there's no difference between using `SetUp`/`TearDown` and the ctor/dtor.

Reply:
Ok, if there is no advantage and it is the idiomatic thing to do, its fine by me :)